pkg/controller/workspace/k8s/utilities.go

```diff
@@ -0,0 +1,38 @@
+package k8s
+
+import (
+	"strings"
+
+	workspaceApi "github.com/che-incubator/che-workspace-crd-operator/pkg/apis/workspace/v1alpha1"
+	. "github.com/che-incubator/che-workspace-crd-operator/pkg/controller/workspace/config"
+	. "github.com/che-incubator/che-workspace-crd-operator/pkg/controller/workspace/model"
+)
+
+func join(sep string, parts ...string) string {
+	return strings.Join(parts, sep)
+}
```

Comment:
The purpose of this function isn't clear to me.

Reply:
Afair the purpose was to make string joining easier.
Moving the separator as first argument allows using variadic params for the string parts to join, and increases readability.

Reply:
Well, it's used for message constructing only. I thought about replacing it with `fmt.Sprintf`, like:
```go
Log.Error(err, Join("", "Cannot find the '", ConfigMapReference.Name, "' ConfigMap in namespace '", ConfigMapReference.Namespace, "'"))
```
vs
```go
Log.Error(err, fmt.Sprintf("Cannot find the '%s' ConfigMap in namespace '%s'", ConfigMapReference.Name, ConfigMapReference.Namespace))
```
WDYT?

Reply:
Well, with Logr that is used here we should not use Sprintf, since it's supposed to print structured logs 
https://github.com/go-logr/logr#why-structured-logging
And then it would look like 
```
Log.Error(err, "ConfigMap does not exist'", "namespace", ConfigMapReference.Namespace, "name", ConfigMapReference.Name)
```

@davidfestal @amisevsk I haven't make myself familiar with that approach but in parallel to do it, I would like ask your opinions - is it better to move to another logger tool and using structured logs is fine.

Reply:
I'm fine with either option, though from what I can tell we're not really making use of the structured aspect of `Logr` very effectively.

Reply:
I leave Logr but used `fmt.Sprintf` for time being.